psiturk/dashboard/__init__.py

```diff
@@ -1,120 +1,133 @@
-from jinja2 import TemplateNotFound
+from __future__ import generator_stop
+from __future__ import annotations
+from flask import Blueprint, render_template, request, current_app as app, \
+    flash, session, g, redirect, url_for
+from flask_login import login_user, logout_user, current_user
 from functools import wraps
-from sqlalchemy import or_
-
 from psiturk.psiturk_config import PsiturkConfig
-from psiturk.experiment_errors import ExperimentError, InvalidUsageError
 from psiturk.user_utils import PsiTurkAuthorization, nocache
-from psiturk.models import Participant
 from psiturk.psiturk_exceptions import *
 
-from psiturk.services_manager import SESSION_SERVICES_MANAGER_MODE_KEY, psiturk_services_manager as services_manager
+from psiturk.services_manager import SESSION_SERVICES_MANAGER_MODE_KEY, \
+    psiturk_services_manager as services_manager
+from flask_login import LoginManager, UserMixin
 
 # # Database setup
-from psiturk.db import db_session, init_db
 from psiturk.models import Participant
-from json import dumps, loads
 
 # load the configuration options
 config = PsiturkConfig()
 config.load_config()
+
 # if you want to add a password protect route use this
 myauth = PsiTurkAuthorization(config)
 
 # import the Blueprint
 dashboard = Blueprint('dashboard', __name__,
-                        template_folder='templates', static_folder='static', url_prefix='/dashboard')
+                      template_folder='templates',
+                      static_folder='static', url_prefix='/dashboard')
 
 
-from flask_login import LoginManager, UserMixin
 login_manager = LoginManager()
 login_manager.login_view = 'dashboard.login'
 
+
 def init_app(app):
     login_manager.init_app(app)
 
+
 class DashboardUser(UserMixin):
     def __init__(self, username=''):
         self.id = username
 
+
 @login_manager.user_loader
 def load_user(username):
     return DashboardUser(username=username)
 
+
 def login_required(view):
     @wraps(view)
     def wrapped_view(*args, **kwargs):
-        if app.login_manager._login_disabled: # for unit testing
+        if app.login_manager._login_disabled:  # for unit testing
             return view(*args, **kwargs)
         is_logged_in = current_user.get_id() is not None
         is_static_resource_call = str(request.endpoint) == 'dashboard.static'
         is_login_route = str(request.url_rule) == '/dashboard/login'
         if not (is_static_resource_call or is_login_route or is_logged_in):
             return login_manager.unauthorized()
         return view(*args, **kwargs)
+
     return wrapped_view
 
+
 def try_amt_services_wrapper(view):
     @wraps(view)
     def wrapped_view(**kwargs):
         try:
-            _ = services_manager.amt_services_wrapper # may throw error
-                                                      # if aws keys not set
+            _ = services_manager.amt_services_wrapper  # may throw error
+            # if aws keys not set
```

Comment:
This second comment row is supposed to be a continuation of the comment immediately above it. shouldn't they be visually aligned?

Reply:
Ah, I think I didn't notice. The PEP8 issue is an in-line comment should have two spaces after the code. Its a little too rigid for how comments end up being used, or for multi-line comments such as this.

Reply:
aligning them makes it complain about unexpected indentation. I'd say we just put in on one line. its super hard to get in-line comments to fit the 80 char limit.